### PR TITLE
Upgrade react-hook-form 7.27.0

### PR DESCRIPTION
**Pull request by alexey-petrov**
* Upgrade react-hook-form to `7.27.0`
* Fixed forms that used react-hook-form and `useArray` hook
* Removed `useArray` usages

### CODE REVIEW

File: src/pages/Account/ProfileQuestions.tsx
Comment (goran-peoski-work) [inline src/pages/Account/ProfileQuestions.tsx:129]: So, keep or delete? A comment like this with no `@debt` explanation doesn't help much.

**Review by goran-peoski-work — CHANGES_REQUESTED** [review]
Some questions, comments, change requests

File: src/components/molecules/AdminUserStatusInput/AdminUserStatusInput.tsx
Comment (alexey-petrov) [reply src/components/molecules/AdminUserStatusInput/AdminUserStatusInput.tsx:42]: it is, it's registered and handled by the form itself

File: src/components/molecules/EntranceStepsInputFieldSet/EntranceStepsInputFieldSet.tsx
Comment (alexey-petrov) [reply src/components/molecules/EntranceStepsInputFieldSet/EntranceStepsInputFieldSet.tsx:30]: yes, now you can just pass control into forms that use `useFieldArray` and that's it

File: src/components/organisms/NewProfileModal/components/links/ProfileModalEditLinks/ProfileModalEditLinks.tsx
Comment (alexey-petrov) [reply src/components/organisms/NewProfileModal/components/links/ProfileModalEditLinks/ProfileModalEditLinks.tsx:43]: renamed to `index`, but not going to use any `link` value since they're dynamic and React not working well when trying to edit a field with a dynamic key

File: src/pages/Account/ProfileQuestions.tsx
Comment (alexey-petrov) [reply src/pages/Account/ProfileQuestions.tsx]: restoredd [9241103](https://github.com/sparkletown/sparkle/pull/2847/commits/9241103be3000f67954e979bcf0e51e9f9b27565)

File: src/components/organisms/WorldGeneralForm/WorldGeneralForm.tsx
Comment (alexey-petrov) [reply src/components/organisms/WorldGeneralForm/WorldGeneralForm.tsx]: [9241103](https://github.com/sparkletown/sparkle/pull/2847/commits/9241103be3000f67954e979bcf0e51e9f9b27565)

File: src/components/organisms/WorldGeneralForm/WorldGeneralForm.tsx
Comment (alexey-petrov) [reply src/components/organisms/WorldGeneralForm/WorldGeneralForm.tsx]: passed as default value [9241103](https://github.com/sparkletown/sparkle/pull/2847/commits/9241103be3000f67954e979bcf0e51e9f9b27565)

File: src/components/organisms/WorldEntranceForm/WorldEntranceForm.tsx
Comment (alexey-petrov) [reply src/components/organisms/WorldEntranceForm/WorldEntranceForm.tsx]: this thing has to accept no arguments to actually `clear` all fields, if used as `clearEntranceSteps removeEntranceStep` it will be accepting `event`, which might interfere with the result that we want to get

File: src/components/molecules/AdminUserStatusInput/AdminUserStatusInput.tsx
Comment (alexey-petrov) [reply src/components/molecules/AdminUserStatusInput/AdminUserStatusInput.tsx]: restored [9241103](https://github.com/sparkletown/sparkle/pull/2847/commits/9241103be3000f67954e979bcf0e51e9f9b27565)

File: src/components/molecules/AdminInput/AdminInput.tsx
Comment (alexey-petrov) [reply src/components/molecules/AdminInput/AdminInput.tsx]: made register required & passed an empty prop in the component above
the reason for it is that one of the fields is always disabled and shouldn't be passed to the form [9241103](https://github.com/sparkletown/sparkle/pull/2847/commits/9241103be3000f67954e979bcf0e51e9f9b27565)

File: src/components/atoms/ImageInput/ImageInput.tsx
Comment (alexey-petrov) [reply src/components/atoms/ImageInput/ImageInput.tsx:36]: so these `any` are used in certain shared components that are used in multiple forms
and if I provide any of those form types others would signal an error
any ideas on how to fix that and have these components understand that they should be able to accept multiple form values?

File: src/components/atoms/InputField/InputField.tsx
Comment (alexey-petrov) [reply src/components/atoms/InputField/InputField.tsx]: [9241103](https://github.com/sparkletown/sparkle/pull/2847/commits/9241103be3000f67954e979bcf0e51e9f9b27565)

File: src/components/atoms/InputField/InputField.tsx
Comment (alexey-petrov) [reply src/components/atoms/InputField/InputField.tsx]: [9241103](https://github.com/sparkletown/sparkle/pull/2847/commits/9241103be3000f67954e979bcf0e51e9f9b27565)

File: src/components/molecules/EntranceStepsInputFieldSet/EntranceStepsInputFieldSet.tsx
Comment (alexey-petrov) [reply src/components/molecules/EntranceStepsInputFieldSet/EntranceStepsInputFieldSet.tsx]: [9241103](https://github.com/sparkletown/sparkle/pull/2847/commits/9241103be3000f67954e979bcf0e51e9f9b27565)

File: src/components/molecules/EntranceStepsInputFieldSet/EntranceStepsInputFieldSet.tsx
Comment (alexey-petrov) [reply src/components/molecules/EntranceStepsInputFieldSet/EntranceStepsInputFieldSet.tsx]: [9241103](https://github.com/sparkletown/sparkle/pull/2847/commits/9241103be3000f67954e979bcf0e51e9f9b27565)

File: src/components/molecules/PollBox/PollBox.tsx
Comment (alexey-petrov) [reply src/components/molecules/PollBox/PollBox.tsx]: [9241103](https://github.com/sparkletown/sparkle/pull/2847/commits/9241103be3000f67954e979bcf0e51e9f9b27565)

File: src/components/molecules/SpaceEditForm/SpaceEditForm.tsx
Comment (alexey-petrov) [reply src/components/molecules/SpaceEditForm/SpaceEditForm.tsx]: [9241103](https://github.com/sparkletown/sparkle/pull/2847/commits/9241103be3000f67954e979bcf0e51e9f9b27565)

File: src/components/molecules/SpaceEditForm/SpaceEditForm.tsx
Comment (alexey-petrov) [reply src/components/molecules/SpaceEditForm/SpaceEditForm.tsx]: [9241103](https://github.com/sparkletown/sparkle/pull/2847/commits/9241103be3000f67954e979bcf0e51e9f9b27565)

File: src/components/molecules/SpaceEditForm/SpaceEditForm.tsx
Comment (alexey-petrov) [reply src/components/molecules/SpaceEditForm/SpaceEditForm.tsx]: [9241103](https://github.com/sparkletown/sparkle/pull/2847/commits/9241103be3000f67954e979bcf0e51e9f9b27565)

File: src/components/molecules/SpaceEditForm/SpaceEditForm.tsx
Comment (alexey-petrov) [reply src/components/molecules/SpaceEditForm/SpaceEditForm.tsx:293]: [9241103](https://github.com/sparkletown/sparkle/pull/2847/commits/9241103be3000f67954e979bcf0e51e9f9b27565)

File: src/components/molecules/SpaceEditForm/SpaceEditForm.tsx
Comment (alexey-petrov) [reply src/components/molecules/SpaceEditForm/SpaceEditForm.tsx]: [9241103](https://github.com/sparkletown/sparkle/pull/2847/commits/9241103be3000f67954e979bcf0e51e9f9b27565)

File: src/components/molecules/SpaceEditForm/SpaceEditForm.tsx
Comment (alexey-petrov) [reply src/components/molecules/SpaceEditForm/SpaceEditForm.tsx]: [9241103](https://github.com/sparkletown/sparkle/pull/2847/commits/9241103be3000f67954e979bcf0e51e9f9b27565)

File: src/components/molecules/SpaceEditForm/SpaceEditForm.tsx
Comment (alexey-petrov) [reply src/components/molecules/SpaceEditForm/SpaceEditForm.tsx:446]: [9241103](https://github.com/sparkletown/sparkle/pull/2847/commits/9241103be3000f67954e979bcf0e51e9f9b27565)

File: src/components/organisms/NewProfileModal/components/ProfileModalChangePassword/ProfileModalChangePassword.tsx
Comment (alexey-petrov) [reply src/components/organisms/NewProfileModal/components/ProfileModalChangePassword/ProfileModalChangePassword.tsx:65]: [9241103](https://github.com/sparkletown/sparkle/pull/2847/commits/9241103be3000f67954e979bcf0e51e9f9b27565)

File: src/components/organisms/SpaceTimingForm/SpaceTimingForm.tsx
Comment (alexey-petrov) [reply src/components/organisms/SpaceTimingForm/SpaceTimingForm.tsx]: [9241103](https://github.com/sparkletown/sparkle/pull/2847/commits/9241103be3000f67954e979bcf0e51e9f9b27565)

File: src/components/organisms/WorldAdvancedForm/WorldAdvancedForm.tsx
Comment (alexey-petrov) [reply src/components/organisms/WorldAdvancedForm/WorldAdvancedForm.tsx]: [9241103](https://github.com/sparkletown/sparkle/pull/2847/commits/9241103be3000f67954e979bcf0e51e9f9b27565)

File: src/components/organisms/WorldAdvancedForm/WorldAdvancedForm.tsx
Comment (alexey-petrov) [reply src/components/organisms/WorldAdvancedForm/WorldAdvancedForm.tsx]: [9241103](https://github.com/sparkletown/sparkle/pull/2847/commits/9241103be3000f67954e979bcf0e51e9f9b27565)

File: src/components/organisms/WorldEntranceForm/WorldEntranceForm.tsx
Comment (alexey-petrov) [reply src/components/organisms/WorldEntranceForm/WorldEntranceForm.tsx]: [9241103](https://github.com/sparkletown/sparkle/pull/2847/commits/9241103be3000f67954e979bcf0e51e9f9b27565)

File: src/components/organisms/WorldAdvancedForm/WorldAdvancedForm.tsx
Comment (alexey-petrov) [reply src/components/organisms/WorldAdvancedForm/WorldAdvancedForm.tsx]: [9241103](https://github.com/sparkletown/sparkle/pull/2847/commits/9241103be3000f67954e979bcf0e51e9f9b27565)

File: src/components/atoms/ImageInput/ImageInput.tsx
Comment (goran-peoski-work) [reply src/components/atoms/ImageInput/ImageInput.tsx:36]: I thought you've seen this thread in due time 
https://doesitsparkle.slack.com/archives/C01ER4M61TL/p1645468187988409?thread_ts=1645442361.700549&cid=C01ER4M61TL 

The problem: if you need to put a comment so the code compiles and deploys, it's no longer a comment.
So, let's make it a code, a simple semi-temporary workaround: 
still have the type effectively be `any`, but remove almost all of those comments that act like code.

File: src/components/organisms/NewProfileModal/components/links/ProfileModalEditLinks/ProfileModalEditLinks.tsx
Comment (goran-peoski-work) [reply src/components/organisms/NewProfileModal/components/links/ProfileModalEditLinks/ProfileModalEditLinks.tsx]: > they're dynamic and React not working well when trying to edit a field with a dynamic key
Well, this is an interesting issue, would be interested to learn more about the problem after we merge this PR

File: src/components/organisms/WorldEntranceForm/WorldEntranceForm.tsx
Comment (goran-peoski-work) [reply src/components/organisms/WorldEntranceForm/WorldEntranceForm.tsx]: just a handful of bad ideas:
- we put `as () => void` after it
- we put `useCallback` on it
- we leave it as it is for now

I had to mention it, but we don't have to waste time looking for a perfect solution.
Let's leave the code as it is.

File: src/components/organisms/WorldGeneralForm/WorldGeneralForm.tsx
Comment (goran-peoski-work) [reply src/components/organisms/WorldGeneralForm/WorldGeneralForm.tsx]: Interesting.
Is validation context now with the new version not necessary?
Does it get picked up if passed through `defaultValues`?

[NOTE] Let's test this scenario after merge.

Comment (goran-peoski-work) [reply src/components/organisms/WorldGeneralForm/WorldGeneralForm.tsx]: I'd still leave a comment after the line, something like
 `// moved from validationContext`
so we'll know how it came to be and fix if there is an issue

File: src/components/organisms/WorldGeneralForm/WorldGeneralForm.tsx
Comment (goran-peoski-work) [inline src/components/organisms/WorldGeneralForm/WorldGeneralForm.tsx:86]: Part of this `worldStartSchema` is directly tied to the old `validationContext: 
```ts
.when(
    "$creating",
    (creating: boolean, schema: Yup.StringSchema) =>
      creating
        ? schema.test(
            "name",
            "This world slug is already taken",
            testWorldBySlugExists
          )
        : schema
  )
```

Is the schema still triggering and doing the proper check?

File: src/components/templates/PosterHall/components/PosterHallSearch/PosterHallSearch.tsx
Comment (goran-peoski-work) [inline src/components/templates/PosterHall/components/PosterHallSearch/PosterHallSearch.tsx:35]: [NOTE]

Because of `register={ALWAYS_NOOP_FUNCTION}` there are few forms that need to be tested. Usually forms wouldn't rely on handlers like `onChange()` and there is probably a way to declare either `onChange` or `register` prop be present, but for now, here is the list:

sparkle-mandatory-register-01

Comment (alexey-petrov) [reply src/components/templates/PosterHall/components/PosterHallSearch/PosterHallSearch.tsx:35]: thanks!

File: src/components/organisms/WorldGeneralForm/WorldGeneralForm.tsx
Comment (alexey-petrov) [reply src/components/organisms/WorldGeneralForm/WorldGeneralForm.tsx:85]: yes, since [56fdacb](https://github.com/sparkletown/sparkle/pull/2847/commits/56fdacb2463fc94ee442760b996bb3d9ba339539)
Talked about it in the slack
```
hm, so after taking a look at that creating form logic I’m still wondering why it existed
since slugs are unique ( otherwise worlds would overlap and not resolve properly ) we can’t have same slugs no matter whether we create or edit a world
so I’m going to add that validation in each of those cases no matter what form is being edited atm
```

File: src/components/molecules/AdminUserStatusInput/AdminUserStatusInput.tsx
Comment (alexey-petrov) [reply src/components/molecules/AdminUserStatusInput/AdminUserStatusInput.tsx]: restored to original `?.` , I've just forgot the way it was before some of my changes
[56fdacb](https://github.com/sparkletown/sparkle/pull/2847/commits/56fdacb2463fc94ee442760b996bb3d9ba339539)

File: src/components/atoms/InputField/InputField.tsx
Comment (alexey-petrov) [reply src/components/atoms/InputField/InputField.tsx]: done [56fdacb](https://github.com/sparkletown/sparkle/pull/2847/commits/56fdacb2463fc94ee442760b996bb3d9ba339539)

**Review by goran-peoski-work — APPROVED** [review]
I think we can merge this if both creating a new world and editing a new world would have the proper validation

File: src/forms/worldStartSchema.ts
Comment (alexey-petrov) [reply src/forms/worldStartSchema.ts:12]: leaving this as a reference to our slack conversation 
image

**Pull request merged**
